### PR TITLE
Explicitely pull managed IAM policies

### PR DESCRIPTION
**Pull request by zhimsel**
https://app.asana.com/0/46565522415617/108217457072054

This may or may not be needed. It seems `iamer` does pull managed
policies, but it does not include non-attached policies. Working on
that.

### CODE REVIEW

**Comment by lra** [comment]
So is it still WIP? Because I tried it and it does not seem to dump new policies that were not dumped before.

FTR: This PR is gonna cover https://github.com/percolate/iamer/issues/3

File: iamer/iammanagedpolicy.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        Returns:
+            unicode
+        """
+        return self._version
+
+    @property
+    def document(self):
+        """
+        Returns:
+            unicode
```
Comment (lra) [inline iamer/iammanagedpolicy.py:45]: dict?

Comment (zhimsel) [inline iamer/iammanagedpolicy.py:45]: You're right. Should be a dict.

**Comment by zhimsel** [comment]
Yeah. It's still a WIP. I noticed that, too. It wasn't pulling any new policies. I have a feeling, however, that it IS actually supporting managed policies, but not directly. If a managed policy is attached to a user/group, it'll get pulled as an in-line policy. So, effectively, as of now it does not support non-attached managed policies.

**Comment by lra** [comment]
but we have a lot of attached managed policies.

**Comment by zhimsel** [comment]
Correct. And it seems the existing functionality (ignoring this PR) already
gets the managed policies, as long as they're attached to something. We
currently only have two non-attached managed policies.

On Thu, Jun 23, 2016 at 3:54 PM, Laurent Raufaste notifications@github.com
wrote:

> but we have a lot of attached managed policies.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/percolate/iamer/pull/4#issuecomment-228196537, or mute
> the thread
> https://github.com/notifications/unsubscribe/AFFDm9o3g-E9oj9VEHgR9P-SKnlt1pZxks5qOwCbgaJpZM4I8WRs
> .

## 

Zach Himsel
Devops Engineer
The must-attend marketing event of the year is back. Get your ticket now
at: percolate.com/transition

**Comment by lra** [comment]
follow up by email

File: iamer/iamcloud.py [hunk 2]
```diff
@@ -212,6 +258,9 @@ def dump_policies(self):
             for policy in group.policies:
                 policies_to_dump.add(policy)
 
+        for managed_policy in self.managed_policies:
+            policies_to_dump.add(policy)
```
Comment (aapalikov) [inline iamer/iamcloud.py]: managed_policy

**Pull request closed without merging**